crates/ra_hir_ty/src/traits/chalk.rs

```diff
@@ -27,6 +27,7 @@ impl chalk_ir::interner::Interner for Interner {
     type InternedGoal = Arc<GoalData<Self>>;
     type InternedGoals = Vec<Goal<Self>>;
     type InternedSubstitution = Vec<Parameter<Self>>;
+    type Identifier = lalrpop_intern::InternedString;
```

Comment:
btw, this means we can now get rid of the lalrpop_intern dependency (and replace this either by `hir::Name` or something interned)